lalrpop/src/normalize/macro_expand/mod.rs

```diff
@@ -53,20 +53,41 @@ impl MacroExpander {
     }
 
     fn expand(&mut self, items: &mut Vec<GrammarItem>) -> NormResult<()> {
-        let mut counter = 0;
+        let mut counter = 0; // Number of items
+        let mut loop_counter = 0;
+
         loop {
             // Find any macro uses in items added since last round and
             // replace them in place with the expanded version:
             for item in &mut items[counter..] {
                 self.replace_item(item);
             }
             counter = items.len();
+            loop_counter += 1;
 
             // No more expansion to do.
             if self.expansion_stack.is_empty() {
                 return Ok(());
             }
 
+            if loop_counter > 200 {
+                // Too much recursion
+                match self.expansion_stack.pop() {
+                    Some(sym) => {
+                        return_err!(
+                            sym.span,
+                            "Exceeded recursion cap while expanding this macro."
+                        );
+                    }
+                    None => {
+                        // This should be impossible, since we just checked is_empty(), but if the
+                        // expansion stack is empty, we're done, so I guess return Ok instead of
+                        // panicing?
+                        return Ok(());
```

Comment:
Hmm, we could use the `unreachable!()` macro though I wouldn't be against unwrapping the option since self.expansion_stack must have at least one element

Reply:
Yeah, I think just unwrap() here does make sense after all.  Pushed a fix for that.